### ISSUE TITLE
pw/pwg literary source display 

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/PWK#96]
This issue is provided as part of explanation of current cdsl display of literary source markup for pw dictionary. The explanation also applies to pwg dictionary.
This explanation is in reference to [this comment](https://github.com/sanskrit-lexicon/PWK/issues/95#issuecomment-1645769053) at issue 95.

The explanation is done via an example from headword akupyant.

## from pw.txt
```
<L>171<pc>1003-1<k1>akupyant<k2>a/kupyant<e>100
{#a/kupyant#}¦ <lex>Adj.</lex> {%nicht aufwallend%} <ls>AV. 29,130,8.</ls>
<LEND>
```

## current display result

<img width="389" alt="image" src="https://github.com/sanskrit-lexicon/PWK/assets/6393033/dc03ecb4-b7be-4ef9-8979-0b42e7ae1b1f">

## relevant html 
This html is generated by the basicdisplay.php

```
<span class="ls">
  <span style="color:blue;" class="ls">Av.</span>
  <span class="ls"> 29,130,8.</span>
</span>
```

## css for ls class
from 'basic.css'
```
.ls {color:gray; font-size:11pt; font-family: charterindocapital;}
```

## line from pw/pywork/pwauth/pwbib_input.txt

```
1020	AV.	Av.	ATHARVAVEDA, Ausg. von ROTH und WHITNEY (ROTH). 
```

**Comment by funderburkjim** [comment]
basicdisplay.php matches the xml element `<ls>AV. 29,130,8.</ls>` with the above line of pwbib_input.txt.
This matches 'AV.' with THE line of pwbib_input.txt using the **second** field (`AV.`)

Once this line is identified,  basicdisplay constructs the html.  Note that the html uses the **third** field (`Av.`).

In rendering the html, the browser displays the abbreviation using the 'ls' css class, which in particular uses
the charterindocapital font.   This font  displays lower-case letters as 'small capital' glyphs.

**Comment by funderburkjim** [comment]
## other dictionaries
The display for other dictionaries (other than pw, pwg) is similar, but not quite the same.
For example, consider the headword anusPura in Grassman:
In gra.txt,
```
<L>542<pc>0061<k1>anusPura<k2>anusPura/
{@(anusphurá),@}¦ <ab>a.</ab>, {%schwirrend%} [<ab>v.</ab> sphur <ab n="mit">m.</ab> ánu].
<div n="TS">-ám śarám <ls>AV. 1,2,3</ls>.
<LEND>
```

<img width="341" alt="image" src="https://github.com/sanskrit-lexicon/PWK/assets/6393033/b4f1549e-3a29-49d4-9b3e-55d0d37c226e">

The matching line of the ls tooltip file for grassman has just 2 fields.
```
AV.	Atharva Veda
```

basicdisplay generates almost the same html as for pw:
```
<span class="ls">
  <span style="color:blue;" class="ls">AV.</span>
  <span class="ls"> 1,2,3</span></span>
```

But notice that we are now displaying the text 'AV.', again with the charerindocapital font; so both the 'A' and the 'V' are capital-letter-sized.

**Comment by funderburkjim** [comment]
The fact that in pwg the 'V' in 'AV.' is displayed as a small capital letter could be considered an error in the 3rd field of the line in pwbib_input.txt.  This could be corrected by changing that line to

```
1020	AV.	AV.	ATHARVAVEDA, Ausg. von ROTH und WHITNEY (ROTH). 
```

**Comment by Andhrabharati** [comment]
@funderburkjim 

My observation is wrt the bib_input file content (in PW family) itself!

The book has the resp. entries (AK, AV, ṚV, TS, VP, etc.) in FULL capital letters, not in small-caps. (Ak, Av, Ṛv, Ts, Vp etc.)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/f26d8b61-5b1c-4a04-9144-d7768a8eae16)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/fe761efb-7b4f-412d-8a8c-4b9a6b3e008f)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/aae0f28c-4b52-448f-99b2-3a23370f33d2)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/9ff778d9-bdae-41a3-adde-be6b2b24ce8c)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/3aceb426-10ec-4b14-a738-5b1ebca9e18b)

**Comment by Andhrabharati** [comment]
> The fact that in pwg the 'V' in 'AV.' is displayed as a small capital letter could be considered an error in the 3rd field of the line in pwbib_input.txt. This could be corrected by changing that line to
> 
> ```
> 1020	AV.	AV.	ATHARVAVEDA, Ausg. von ROTH und WHITNEY (ROTH). 
> ```

Exactly; and, I had posted the PWG file 2 years back with such corrections (see my above post) which got ignored thus far.

**Comment by funderburkjim** [comment]
Should we have that 3rd field at all?

The print of PW does *often* (but not always) use large/small capital letters.
e.g.
<img width="332" alt="image" src="https://github.com/sanskrit-lexicon/PWK/assets/6393033/49eb93a2-b8cb-491b-a72c-3452e17f2644">

<img width="185" alt="image" src="https://github.com/sanskrit-lexicon/PWK/assets/6393033/8244ecc9-315d-4eb6-9a98-326f5267a687">

And current cdsl displays mimic the print fairly well (except for the capitalization errors in pwbib_input.txt such as for AV)
<img width="443" alt="image" src="https://github.com/sanskrit-lexicon/PWK/assets/6393033/7b536dfd-685f-407c-bd54-121f3679eefa">

But sometimes the print uses more normal letters for the ls abbreviation.  Such as for 'Spr.' 
<img width="349" alt="image" src="https://github.com/sanskrit-lexicon/PWK/assets/6393033/b5c9c1f2-7829-4583-9ad8-a6ef46ff3a2f">

So for cases like 'Spr.' , the cdsl display does NOT mimic the print.

**Comment by funderburkjim** [comment]
So, you would be in favor of changing pwbib_input.txt to the form of Grassman?

i.e. dropping the 1st and 3rd fields. example:
```
AV.  ATHARVAVEDA, Ausg. von ROTH und WHITNEY (ROTH). 

```

**Comment by Andhrabharati** [comment]
If agreed upon, I can help preparing the bib_input files for PWG and pwk, as in every other CDSL work.

In fact, at the end of my current pwk (re-)working, I had planned to give out my resolved ls-entries, which I had kept on hold thus far (though prepared 2 years back itself).

Now, as I am looking further into pwk, finding more ls-entries and also some corrections to the earlier ones marked. 
[Last time, I had spent just two days on the pwk; but stopped abruptly.]

**Comment by Andhrabharati** [comment]
> So, you would be in favor of changing pwbib_input.txt to the form of Grassman?
> 
> i.e. dropping the 1st and 3rd fields. example:
> 
> ```
> AV.  ATHARVAVEDA, Ausg. von ROTH und WHITNEY (ROTH). 
> ```

Not just that, I would even be splitting some of them further depending on context; you may note that the print has clubbed two or more separable entries together under a single biblio entry at few places.

Say, for AV

**Comment by funderburkjim** [comment]
> If agreed upon, I can help preparing the bib_input files for  pwk,

Agreed.  

A copy of the current pwbib_input.txt file is in the issue95/litsrc directory of as pwbib_input_0.txt.

When you modify this,  I will need to correspondingly modify
* pywork code to create the sqlite version (pwbib.sqlite) of the tooltip file from the revised pwbib_input.txt
  * https://github.com/sanskrit-lexicon/csl-pywork/tree/master/v02/distinctfiles/pw/pywork/pwauth
* basicdisplay program to use the revised 2-field format of that sqlite file
